test/test_add_page.py

```diff
@@ -29,3 +29,19 @@ def test_add_page_duration(tmp_path):
     pdf.add_page()
     pdf.cell(txt="Page 3")
     assert_pdf_equal(pdf, HERE / "add_page_duration.pdf", tmp_path)
+
+
+def test_break_or_add_page(tmp_path):
+    pdf = fpdf.FPDF()
+    pdf.set_auto_page_break(auto=True)
+    pdf._break_or_add_page()
+    pdf.set_font("Helvetica", "", 16)
+    for i in range(1, 51):
+        pdf.set_x(10)
+        pdf.multi_cell(50, 10, f"Text {i} - Page {pdf.page}", 1, "C")
+    pdf.page = 1
```

Comment:
It's a very interesting test overall, thanks for submitting it!

We currently have absolutely zero test covering this usage of "going back to a previous page to generate stuff".
Actually, some `fpdf2` internal mechanisms may break by doing this...
More precisely: in `_putpages` & `_substitute_page_number`, `nb` should be defined as `len(self.pages)`.
Could you please fix this as part of this PR?

To be clear: **moving back to a previous page to write things may not currently be well supported by `fpdf2`**
I'd be inclined to implement this "right" though, as `fpdf.FPDF` does not currently forbid users to do so...